### ISSUE TITLE
Notebook guide for beginners

### ISSUE THREAD

**Issue by tritemio** [jupyter/jupyter#3]
I'm opening this issue to follow-up a recent discussion on ipython-dev ([thread](http://mail.scipy.org/pipermail/ipython-dev/2015-June/016463.html)).

Briefly, I think there is the need for a concise documentation for absolute beginners that receive a notebook and need to run it. My main use-case is the notebook-as-an-app in which we provide notebooks as mini-apps to perform a single task. The user needs only to know how to run it and change a few parameters (i.e. input file name).

I had hard time finding docs for beginners so I started such a guide:

http://jupyter-notebook-beginner-guide.readthedocs.org/

and @Carreau suggested to open an issue here to discuss possible integration with the official documentation.

**Comment by Carreau** [comment]
Pinging @ellisonbg explicitely, I know you are grading, but as you have a couple of student that might go through the docs it might be nice to coordinate here.

**Comment by ellisonbg** [comment]
A few short comments:

* We are doing a lot of work on our documentation right now and most of the content that is in this beginner guide will be in our docs in some form.
* One of the active areas of work is in converting our docs to reflect the jupyter branding/organization. All of that will be in the 4.0 docs.
* The beginner guide linked to above is a mix of very general things (like the glossary) and more specific things. I don't think that exact mix will work well for our general documentation, which needs to have a consistent level of detail. However a good part of that content, perhaps organized differently, should end up in our docs for 4.0.

Summary: I don't think it will make sense to outright merge the full beginner guide into our docs, but parts of it may be appropriate to integrate in places. Let's leave this open and as we start to outline the docs work for 4.0 we can see where this might fit in.

**Comment by Carreau** [comment]
I think the issue is also mostly to help with docs in general, not to integrate the beginner guide as is. 
I don't see any reason either to have a section on the sidebar which is beginner guide.

**Comment by tritemio** [comment]
The issue is that this guide should no exists ;). It does because of a current lack in the docs.

I understand that you are going to reshape the full docs and websites and that's good. But Jupyter is a big project and structure is a key, you don't want to overwhelm user with too much information at first glance (not implying you are doing that, just bringing up the point).

Which users? This depends on your vision. My small use-case, that I'm advocating for, is of a user wanting to reproduce a simple notebook attached to a publication. This user does not know anything about python or the notebook but he has a goal, and it may be not inclined spending half a day to understand how to setup and run things. He may be not interested in parallel or cloud computing. He just wants quick and terse instructions to reach the goal. Once he is struck with awe, he may go back and read more about the "big picture" ;). 

What's amazing about ipython/jupyter is that it is realistic, given targeted instructions, for a new user to go from 0 to running a notebook on your desktop in 20 minutes. Targeted instructions is the key: there are many practical details the user needs to know to get started all these should be all in one easily accessible place.

So my question to you (ipython devs) is: how important is this class of user to you? Is it all going to bee MOOC and distributed notebooks :)? I think the desktop will stay relevant for a while and targeting these users would expand your traditional user-base to a less "tech-savy" one, which is good!

Or maybe I'm stating the obvious, so bear with me :).

**Comment by Carreau** [comment]
Ok, seeing that the "lot of work on documentation" have staled for a bit in this repo may I suggest the following. @tritemio do you want to make a PR with a section "Absolute Beginner guide to Jupyter Notebook" ? Once it's in, it;s easier to move pieces around, than to do it cross repos ?

I prefer to have moving docs published every couple of days, that changes, than to bikeshed weeks on somethings that won't be done.

**Comment by ellisonbg** [comment]
Sorry, getting back to things after being sick for 5 days.

The main issue I see is that one of the central problems our docs have is
that we have lots of overlapping content, that says the same thing, at
different levels of depth and granularity - with some of the versions being
outdated or wrong. Because of this our docs will not, and cannot improve,
through mere additions. To improve them, we need to start creating a
coherent conceptual organization for our docs, with a consistent level of
granularity, and no major duplication of content.

Much of the content in the beginners guide linked to here is already in
various places on our docs. If you want to have a go at integrating what
you have done in a coherent manner, and deduplicating things in a way that
fixes the underlying organization issues that would be great.

We are also having a thread on the mailing list about documentation - I
would link to it, but Google Groups is angry with me right now...I bring
that up because we may have some discussions about docs at SciPy that will
set the direction of our strategy with docs. All of that will be posted on
the jupyter google group as it happens.

On Fri, Jun 19, 2015 at 7:17 AM, Antonino Ingargiola <
notifications@github.com> wrote:

> The issue is that this guide should no exists ;). It does because of a
> current lack in the docs.
>
> I understand that you are going to reshape the full docs and websites and
> that's good. But Jupyter is a big project and structure is a key, you don't
> want to overwhelm user with too much information at first glance (not
> implying you are doing that, just bringing up the point).
>
> Which users? This depends on your vision. My small use-case, that I'm
> advocating for, is of a user wanting to reproduce a simple notebook
> attached to a publication. This user does not know anything about python or
> the notebook but he has a goal, and it may be not inclined spending half a
> day to understand how to setup and run things. He may be not interested in
> parallel or cloud computing. He just wants quick and terse instructions to
> reach the goal. Once he is struck with awe, he may go back and read more
> about the "big picture" ;).
>
> What's amazing about ipython/jupyter is that it is realistic, given
> targeted instructions, for a new user to go from 0 to running a notebook on
> your desktop in 20 minutes. Targeted instructions is the key: there are
> many practical details the user needs to know to get started all these
> should be all in one easily accessible place.
>
> So my question to you (ipython devs) is: how important is this class of
> user to you? Is it all going to bee MOOC and distributed notebooks :)? I
> think the desktop will stay relevant for a while and targeting these users
> would expand your traditional user-base to a less "tech-savy" one, which is
> good!
>
> Or maybe I'm stating the obvious, so bear with me :).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter/issues/3#issuecomment-113528618>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by tritemio** [comment]
@ellisonbg

> To improve them, we need to start creating a
> coherent conceptual organization for our docs, with a consistent level of
> granularity, and no major duplication of content.

Completely agree.

> Much of the content in the beginners guide linked to here is already in
> various places on our docs. If you want to have a go at integrating what
> you have done in a coherent manner, and deduplicating things in a way that
> fixes the underlying organization issues that would be great.

I took a look at the official docs, and even ignoring that is not adapted for newbies does not contain "most" of the content in this beginner guide. IMHO, there is the need for an "absolute beginner guide" that is self-contained and detailed enough to bring a newbie from 0 to running a notebook in a few minutes. It should also be as short as possible, with links to deeper docs.

Please take a look at the latest version of the guide. There are 3 sections:

## What is Jupyter Notebook

This contains a 1-paragraph description for each of:

1. notebook document
2. jupyter notebook application
3. kernel
4. dashboard

Each paragraphs ends with the most relevant link I found at this moment to the official docs.

## Installation

Just mentioning to download and install Anaconda. The official docs installation page is not for newbies.

## Run Jupyter

It contains how to start the server, change the startup folder, open notebooks, shutdown a kernel or the whole Jupyter Notebook application and how to run a notebook. Plus some warning about opening the same notebooks in multiple tabs, multiple instances of Jupyter Notebook, autosaving and copying.

Most of this content is not covered in the official docs.

----
Where possible, I tried to link to official docs, that is currently too detailed or with gaps about basic information.

Bottom line, even improving the official docs by filling the gaps (or by deep refactoring), there is the need for a beginner tutorial that with all the info in one place. The few paragraphs of "duplication" are necessary.

@Carreau, regarding a possible PR, seeing the readthedocs page in the jupyter repository I don't think this is the right place for such a beginner guide. Currently this is a developer resource.

**Comment by Carreau** [comment]
The main page at Jupyter.readthedocs.org should be able to redirect people to the right place. At least the front page. 

[Julia](http://docs.julialang.org/en/release-0.3/) does a pretty good job at that, and mix dev and beginner user docs in a single place.

Seeing that the repo has not moved for a month now, any change can only be improvement.So feel free to send PRs.

**Comment by takluyver** [comment]
I also quite like the style of the [Python docs landing page](https://docs.python.org/3/). I don't think we should copy that exactly, but it does a good job of dispatching to different places for some very different interests.

**Comment by ellisonbg** [comment]
I talked with Min and I am still planning on finishing the jupyter landing
pages for docs for the 4.0 release.

On Thu, Jul 16, 2015 at 11:17 AM, Thomas Kluyver <notifications@github.com>
wrote:

> I also quite like the style of the Python docs landing page
> <https://docs.python.org/3/>. I don't think we should copy that exactly,
> but it does a good job of dispatching to different places for some very
> different interests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter/issues/3#issuecomment-122039725>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by tritemio** [comment]
Also, as much as I like the RTD theme, I think Jupyter should have it's own theme. It is too easy to get confused between other project docs and official jupyter docs, especially for beginners.

**Comment by minrk** [comment]
@tritemio I actually disagree with that. I think Jupyter *should* appear as just another project, among the many on RTD.

**Comment by Carreau** [comment]
>> Also, as much as I like the RTD theme, I think Jupyter should have it's own theme. It is too easy to get confused between other project docs and official jupyter docs, especially for beginners.

> @tritemio I actually disagree with that. I think Jupyter should appear as just another project, among the many on RTD.

Conda [does a pretty good job](http://conda.pydata.org/docs/faq.html)  as just slightly changing the colorscheme.

**Comment by tritemio** [comment]
@minrk why? Helping to understand which project is which is good.

@Carreau, yes a subtle color scheme change I think is enough.

**Comment by willingc** [comment]
@minrk @ellisonbg I love writing docs as much as I love coding. Great documentation empowers individuals to be productive quickly with minimal frustration. Is there a good link for me to get up to speed on the doc plans for 4.0 and general doc strategy for the Jupyter ecosystem?

**Comment by Carreau** [comment]
Hey @willingc , roughly this repo will be a high level doc that redirect on each subproject page. Each repo will have it's own pages on readthedocs. We plan on trying to get docs as notebook also. 

The is also some information on our meetings notes from [June 9th](https://ipython.hackpad.com/2015-Dev-Meetings-Part-1-3YJG5lv2Hws#:h=June-9)

**Comment by Jeffiy** [comment]
Hey, jupyter can't use like this "jupyter notebook --profile=nb", is there a replacement for "ipython notebook --profile=nb",

**Comment by minrk** [comment]
@Jeffiy you can use a custom jupyter dir with `JUPYTER_CONFIG_DIR=/path/to/nondefault/config jupyter notebook`

There is no longer a way to specify both the IPython kernel's profile and the Jupyter notebook's config directory on the jupyter command-line.

**Comment by Jeffiy** [comment]
@minrk thank you, but everytime I start notebook I must type so long command,it's not cool any more.

**Comment by damianavila** [comment]
You don't need to type the long command every time if you, for instance, set up the `JUPYTER_CONFIG_DIR` env variable in your `.bashrc` file (more info about how to do it in this article: https://www.digitalocean.com/community/tutorials/how-to-read-and-set-environmental-and-shell-variables-on-a-linux-vps).

**Comment by Jeffiy** [comment]
@damianavila Thank you.

**Comment by willingc** [comment]
Closing this issue since docs for beginners have improved in the past 6 months. Please feel free to open a specific issue if there are things that are still desired for beginners. :surfer:

**Comment by tritemio** [comment]
@willingc, the docs had great improvements, thanks!

I still refer users of my software to the jupyter quick start guide linked at the beginning of this issue because the general documentation is necessarily more complex. The jupyter quick start guide covers only one use case (local single-user installation of the notebook) so it can be simpler, and that's a bonus for totally new users.
